src/main/kotlin/app/widgets/ProgressBar.kt

```diff
@@ -0,0 +1,50 @@
+package app.ui
+
+
+import javafx.application.Platform
+import javafx.geometry.Pos
+import javafx.scene.layout.HBox
+import javafx.scene.layout.VBox
+
+
+import tornadofx.*
+import java.util.*
+import kotlin.concurrent.thread
+
+/**
+ * sweet code for a heckin cool loading bar
+ */
+class ProgressBar : VBox() {
+
+    init {
+
+        importStylesheet("/progressbar.css")
+        var messages = ResourceBundle.getBundle("MyView")
+
+        var isItDone = false
```

Comment:
because you update this value and check it in different threads, one of the threads may end up pulling a cached (dirty) value; as a result you need to have this field annotated as @Volatile